assets/js/components/Dashboard/ScreenDetail.tsx

```diff
@@ -11,18 +11,30 @@ interface ScreenDetailProps {
 }
 
 const ScreenDetail = (props: ScreenDetailProps): JSX.Element => {
-  const translateScreenType = () => {
-    return SCREEN_TYPES.find(({ ids }) => ids.includes(props.screens[0].type))
-      ?.label;
-  };
+  const isSolari = props.screens.every((screen) => screen.type === "solari");
 
-  const isPaess = props.screens.every((screen) => screen.type === "pa_ess");
+  return isSolari ? (
+    <div className="screen-detail__solari-layout">
+      {props.screens.map((screens, index) => (
+        <ScreenCard {...props} key={index} screens={[screens]} />
+      ))}
+    </div>
+  ) : (
+    <ScreenCard {...props} />
+  );
+};
 
-  const getPaessRouteLetter = () => {
-    return props.screens[0].station_code
-      ? props.screens[0].station_code.charAt(0).toLowerCase()
-      : "";
-  };
+const ScreenCard = (props: ScreenDetailProps) => {
+  const { screens, isOpen } = props;
+  const isPaess = screens.every((screen) => screen.type === "pa_ess");
+  const isSolari = screens.every((screen) => screen.type === "solari");
+  const paessRouteLetter = screens[0].station_code
+    ? screens[0].station_code.charAt(0).toLowerCase()
+    : "";
+
+  const translateScreenType = SCREEN_TYPES.find(({ ids }) =>
```

Comment:
Super nitpicky, but the name of this still reads like it is a function. What do you think about `translatedScreenType`?

Reply:
oh yeah, definitely worth mentioning! thanks